components/collections/src/codepointinvliststringlist/mod.rs

```diff
@@ -0,0 +1,310 @@
+            str_list,
+        })
+    }
+
+    #[doc(hidden)]
+    pub const fn from_parts_unchecked(
+        cp_inv_list: CodePointInversionList<'data>,
+        str_list: VarZeroVec<'data, str>,
+    ) -> Self {
+        CodePointInversionListStringList {
+            cp_inv_list,
+            str_list,
+        }
+    }
+
+    /// Returns the number of elements in this set (its cardinality).
+    /// Note than the elements of a set may include both individual
+    /// codepoints and strings.
+    pub fn size(&self) -> usize {
+        self.cp_inv_list.size() + self.str_list.len()
+    }
+
+    /// Return true if this set contains multi-code point strings or the empty string.
+    pub fn has_strings(&self) -> bool {
+        !self.str_list.is_empty()
+    }
+
+    ///
+    /// # Examples
+    /// ```
+    /// use icu_collections::codepointinvlist::CodePointInversionList;
+    /// use icu_collections::codepointinvliststringlist::CodePointInversionListStringList;
+    /// use zerovec::VarZeroVec;
+    ///
+    /// let cp_slice = &[0, 0x0080, 0xFFFF, 0x1_0000, 0x10_FFFF, 0x11_0000];
+    /// let cp_list =
+    ///    CodePointInversionList::try_clone_from_inversion_list_slice(cp_slice).unwrap();
+    /// let str_slice = &["", "ascii_max", "bmp_max", "unicode_max", "zero"];
+    /// let str_list = VarZeroVec::<str>::from(str_slice);
+    ///
+    /// let cpilsl = CodePointInversionListStringList::try_from(cp_list, str_list).unwrap();
+    ///
+    /// assert!(cpilsl.contains("ascii_max"));
+    /// assert!(cpilsl.contains(""));
+    /// assert!(cpilsl.contains("A"));
+    /// assert!(!cpilsl.contains("bazinga!"));
+    /// ```
+    pub fn contains(&self, s: &str) -> bool {
+        if s.len() == 1 {
+            if let Some(ch) = s.chars().next() {
+                self.contains_char(ch)
+            } else {
+                false
+            }
+        } else {
+            #[allow(clippy::filter_next)]
+            // VarZeroVec's iterator supports .filter() but not .find()
+            self.str_list.iter().filter(|&x| x == s).next().is_some()
+        }
+    }
+
+    ///
+    /// # Examples
+    /// ```
+    /// use icu_collections::codepointinvlist::CodePointInversionList;
+    /// use icu_collections::codepointinvliststringlist::CodePointInversionListStringList;
+    /// use zerovec::VarZeroVec;
+    ///
+    /// let cp_slice = &[0, 0x80, 0xFFFF, 0x1_0000, 0x10_FFFF, 0x11_0000];
+    /// let cp_list =
+    ///     CodePointInversionList::try_clone_from_inversion_list_slice(cp_slice).unwrap();
+    /// let str_slice = &["", "ascii_max", "bmp_max", "unicode_max", "zero"];
+    /// let str_list = VarZeroVec::<str>::from(str_slice);
+    ///
+    /// let cpilsl = CodePointInversionListStringList::try_from(cp_list, str_list).unwrap();
+    ///
+    /// assert!(cpilsl.contains_u32(0));
+    /// assert!(cpilsl.contains_u32(0x0042));
+    /// assert!(!cpilsl.contains_u32(0x0080));
+    /// ```
+    pub fn contains_u32(&self, cp: u32) -> bool {
```

Comment:
Nit: let's stick with the `contains32` convention I think. Right?

Reply:
Done. Yes, that is consistent.